pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -70,61 +71,92 @@ func (c *Controller) ConfigDescriptor() model.ConfigDescriptor {
 }
 
 // List returns all the config that is stored by type and namespace
-func (c *Controller) List(typ, namespace string) ([]model.Config, error) {
+// if namespace is empty string it returns config for all the namespaces
+func (c *Controller) List(typ, namespace string) (out []model.Config, err error) {
 	_, ok := c.ConfigDescriptor().GetByType(typ)
 	if !ok {
 		return nil, fmt.Errorf("list unknown type %s", typ)
 	}
 	data, exists := c.configStore[typ]
-	if !exists {
-		log.Infof("list config not found for the type %s\n", typ)
-		return nil, nil
+	if exists && len(data) == 0 {
+		return out, nil
 	}
-	out := make([]model.Config, 0, len(c.configStore[typ]))
+
 	if namespace == "" {
-		for _, ns := range data {
-			ns.Range(func(key, value interface{}) bool {
-				out = append(out, value.(model.Config))
-				return true
-			})
+		c.configStoreMu.Lock()
+		byType, ok := c.configStore[typ]
+		c.configStoreMu.Unlock()
```

Comment:
This locking looked incorrect until I remembered the per-type store is  effectively immutable due to the `apply` logic below. Can you add a comment to remind others (myself included :)) why accessing the inner store is safe without a lock?